common/src/main/java/com/wynntils/screens/maps/MainMapScreen.java

```diff
@@ -262,8 +265,29 @@ private void renderPois(PoseStack poseStack, int mouseX, int mouseY) {
                                 /*|| (hadesUser.isGuildMember() && Managers.Feature.getFeatureInstance(MapFeature.class).renderRemoteGuildPlayers)*/ )
                         .map(PlayerMainMapPoi::new));
 
-        if (KeyboardUtils.isControlDown()) {
+        if (Managers.Feature.getFeatureInstance(MapFeature.class)
```

Comment:
Let's move this whole logic to key events instead of render:

```
keyDown:
if showTerrs is true and toggleMode is true -> showTerrs is false
otherwise -> showTerrs is true
```

```
keyUp:
if showTerrs is true and toggleMode is false -> showTerrs is false
````

Please override these methods and use this, it is much easier to read.